src/main/scala/sparkz/core/api/http/PeersApiRoute.scala

```diff
@@ -80,12 +117,47 @@ case class PeersApiRoute(peerManager: ActorRef,
     }
   }
 
-  def blacklistedPeers: Route = (path("blacklisted") & get) {
-    val result = askActor[Seq[InetAddress]](peerManager, GetBlacklistedPeers)
-      .map(x => BlacklistedPeers(x.map(_.toString)).asJson)
-    ApiResponse(result)
+  def addToBlacklist: Route = (path("blacklist") & post & withAuth & entity(as[Json])) { json =>
+    val maybeAddress = json.asString.flatMap(addressAndPortRegexp.findFirstMatchIn)
+
+    maybeAddress match {
+      case None => ApiError(StatusCodes.BadRequest, s"address $maybeAddress is not well formatted")
+
+      case Some(addressAndPort) =>
+        val host = InetAddress.getByName(addressAndPort.group(1))
+        val port = addressAndPort.group(2).toInt
+        peerManager ! AddToBlacklist(new InetSocketAddress(host, port))
```

Comment:
same here as below, do we need to disconnect the peer if was connected?
I think, here we sure need.

Reply:
Here we do not need to disconnect, once we send the AddToBlacklist msg to the peer manager it already sends a CloseConnection msg to the NetworkController